vita3k/app/src/app_init.cpp

```diff
@@ -168,37 +105,28 @@ bool init(HostState &state, Config cfg, const Root &root_paths) {
     else
         state.pref_path = state.cfg.pref_path + '/';
 
-    state.window = WindowPtr(SDL_CreateWindow(window_title, SDL_WINDOWPOS_CENTERED, SDL_WINDOWPOS_CENTERED, DEFAULT_RES_WIDTH, DEFAULT_RES_HEIGHT, SDL_WINDOW_OPENGL | SDL_WINDOW_RESIZABLE), SDL_DestroyWindow);
-    if (!state.window || !init(state.mem) || !init(state.audio, resume_thread) || !init(state.io, state.base_path, state.pref_path)) {
-        return false;
-    }
-
-    // Recursively create GL version until one accepts
-    // Major 4 is mandantory
-    const int accept_gl_version[] = {
-        5, // OpenGL 4.5
-        3, // OpenGL 4.3
-        2, // OpenGL 4.2
-        1 // OpenGL 4.1
-    };
-
-    SDL_GL_SetAttribute(SDL_GL_CONTEXT_MAJOR_VERSION, 4);
-    SDL_GL_SetAttribute(SDL_GL_CONTEXT_PROFILE_MASK, SDL_GL_CONTEXT_PROFILE_CORE);
-
-    int choosen_minor_version = 0;
-
-    for (int i = 0; i < sizeof(accept_gl_version) / sizeof(int); i++) {
-        SDL_GL_SetAttribute(SDL_GL_CONTEXT_MINOR_VERSION, accept_gl_version[i]);
+    renderer::Backend backend;
+#ifdef USE_VULKAN
+    if (state.cfg.backend == "Vulkan")
+        backend = renderer::Backend::Vulkan;
+    else
+#endif
+        backend = renderer::Backend::OpenGL;
 
-        state.glcontext = GLContextPtr(SDL_GL_CreateContext(state.window.get()), SDL_GL_DeleteContext);
-        if (state.glcontext) {
-            choosen_minor_version = accept_gl_version[i];
-            break;
-        }
+    SDL_WindowFlags window_type;
+    switch (backend) {
```

Comment:
Default case missing! In this case you should tell bad configuration happens. I want to keep this switch if we ever have future backend, so add a failure on default

Reply:
Done.